synapse/storage/room.py

```diff
@@ -323,6 +324,7 @@ def _store_event_search_txn(self, txn, event, key, value):
                     event.origin_server_ts,
                 )
             )
+            txn.execute("RESET work_mem")
```

Comment:
what if the insert throws an exception? do we not need to reset work_mem in that case